exporter/containerimage/writer.go

```diff
@@ -261,15 +265,16 @@ func (ic *ImageWriter) Commit(ctx context.Context, inp *exporter.Source, session
 			inlineCacheEntry, _ = inlineCacheResult.FindRef(p.ID)
 		}
 
-		desc, _, err := ic.commitDistributionManifest(ctx, opts, r, config, remote, opts.Annotations.Platform(&p.Platform), inlineCacheEntry, opts.Epoch, session.NewGroup(sessionID), baseImg)
+		mfstDesc, configDesc, err := ic.commitDistributionManifest(ctx, opts, r, config, remote, opts.Annotations.Platform(&p.Platform), inlineCacheEntry, opts.Epoch, session.NewGroup(sessionID), baseImg)
 		if err != nil {
 			return nil, err
 		}
 		dp := p.Platform
-		desc.Platform = &dp
-		idx.Manifests = append(idx.Manifests, *desc)
+		mfstDesc.Platform = &dp
+		idx.Manifests = append(idx.Manifests, *mfstDesc)
+		descriptors = append(descriptors, mfstDesc, configDesc)
```

Comment:
Shouldn't this be just `mfstDesc`. Mixing manifest and config descriptors together does not make a predictable result. And how would one even connect the ones that are for the same image.